platform/ext/target/nordic_nrf/nrf5340/spm_hal.c

```diff
@@ -46,9 +48,49 @@ enum tfm_plat_err_t tfm_spm_hal_configure_default_isolation(
                   uint32_t partition_idx,
                   const struct tfm_spm_partition_platform_data_t *platform_data)
 {
-    /* FIXME we need to see how to handle this because nRF IDAU peripheral
-     * is not able to assign PRIV/nPRIV access policy configuration
+#if defined(CONFIG_TFM_ENABLE_MEMORY_PROTECT) && (TFM_LVL != 1)
+    bool privileged = tfm_is_partition_privileged(partition_idx);
+    struct mpu_armv8m_region_cfg_t region_cfg;
+#endif
+
+    if (!platform_data) {
+        return TFM_PLAT_ERR_INVALID_INPUT;
+    }
+
+#if defined(CONFIG_TFM_ENABLE_MEMORY_PROTECT) && (TFM_LVL != 1)
+    if (!privileged) {
+        region_cfg.region_nr = PARTITION_REGION_PERIPH_START +
```

Comment:
`struct tfm_spm_partition_platform_data_t`, `PARTITION_REGION_PERIPH_START` and `periph_num_count` look undefined.

Reply:
That's fixed now, thanks.

Reply:
Does it work when they are defined after the usage?

Reply:
Nope. I had to re-shuffle that. Also I had to include region.h (this came from last upmerge)
Done now.